cmd/root.go

```diff
@@ -102,21 +102,32 @@ func init() {
 func initConfig() {
 	userHome := utils.GetHomeDirectory()
 	defaultPath := path.Join(userHome, whisk.DEFAULT_LOCAL_CONFIG)
-	if utils.Flags.CfgFile != "" {
 
-		// Read the file as a wskprops file, to check if it is valid.
-		_, err := whisk.ReadProps(utils.Flags.CfgFile)
-		if err != nil {
-			utils.Flags.CfgFile = defaultPath
-			warn := wski18n.T(wski18n.ID_WARN_CONFIG_INVALID_X_path_X,
-				map[string]interface{}{
-					wski18n.KEY_PATH: utils.Flags.CfgFile})
-			wskprint.PrintOpenWhiskWarning(warn)
+    // Precedence order for reading the configuration file should be:
+    // 1. --config 
+    // 2. ENV variable WSK_CONFIG_FILE
+    // 3. Default $HOME/.wskprops
+	cfgFiles := []string{
+		utils.Flags.CfgFile,
+		os.Getenv("WSK_CONFIG_FILE"),
+        defaultPath,
+	 }
+
+	for _, cfgFile := range cfgFiles {
```

Comment:
If indeed there are 3 discrete places we look for a valid config. file (and each should have a single entry/key and associated value), then using an array to loop upon with a break stmt. seems overkill.

Reply:
validating each in order and discontinuing search (short circuit) on first success seems much more efficient.

Reply:
Well, I am not a gopher, therefore I do not know if there is any better method to perform it. What would you suggest to do?

Reply:
Can you please clarify "not a gopher" for me?

Reply:
I mean that I an not familiar with Go, therefore I do not know if there any more efficient ways of reading the configuration files than trying to read them in order and breaking the loop on first success.